File: src/test/java/cube8540/oauth/authentication/credentials/oauth/client/infra/rule/DefaultOAuth2ClientOwnerValidationRuleTest.java
package cube8540.oauth.authentication.credentials.oauth.client.infra.rule;

import cube8540.oauth.authentication.credentials.oauth.client.domain.OAuth2Client;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@DisplayName("기본 클라이언트 소유자 유효성 검사 클래스 테스트")
class DefaultOAuth2ClientOwnerValidationRuleTest {

    private DefaultOAuth2ClientOwnerValidationRule rule;

    @BeforeEach
    void setup() {
        this.rule = new DefaultOAuth2ClientOwnerValidationRule();
    }

    @Test
    @DisplayName("클라이언트의 소유자가 null 일시 유효성 검사 결과는 false 가 반환되여야 한다.")
    void clientOwnerNullValidationShouldReturnsFalse() {
        OAuth2Client client = mock(OAuth2Client.class);

        when(client.getOwner()).thenReturn(null);

        assertFalse(rule.isValid(client));
    }

}

File: src/test/kotlin/cube8540/oauth/authentication/oauth/client/infra/DefaultOAuth2ClientOwnerValidationRuleTest.kt
package cube8540.oauth.authentication.oauth.client.infra

import cube8540.oauth.authentication.oauth.client.domain.OAuth2Client
import io.mockk.every
import io.mockk.mockk
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class DefaultOAuth2ClientOwnerValidationRuleTest {

    private val rule = DefaultOAuth2ClientOwnerValidationRule()

    @Test
    fun `client owner must not be null`() {
        val client: OAuth2Client = mockk {
            every { owner } returns null
        }

        val result = rule.isValid(client)
        assertThat(result).isFalse
    }

}